include/velocypack/Builder.h

```diff
@@ -546,22 +647,29 @@ class Builder {
   }
 
   template <typename T>
-  uint8_t* addInternal(std::string const& attrName, T const& sub) {
-    return addInternal<T>(attrName.data(), attrName.size(), sub);
+  uint8_t* addInternal(std::string const& attrName, uint64_t tag, T const& sub) {
+    return addInternal<T>(attrName.data(), attrName.size(), tag, sub);
   }
 
   template <typename T>
-  uint8_t* addInternal(StringRef const& attrName, T const& sub) {
-    return addInternal<T>(attrName.data(), attrName.size(), sub);
+  uint8_t* addInternal(StringRef const& attrName, uint64_t tag, T const& sub) {
+    return addInternal<T>(attrName.data(), attrName.size(), tag, sub);
   }
 
   template <typename T>
-  uint8_t* addInternal(char const* attrName, T const& sub) {
-    return addInternal<T>(attrName, strlen(attrName), sub);
+  uint8_t* addInternal(char const* attrName, uint64_t tag, T const& sub) {
+    // null check for array creation
+    return addInternal<T>(attrName, attrName == nullptr ? 0 : strlen(attrName), tag, sub);
   }
 
   template <typename T>
-  uint8_t* addInternal(char const* attrName, std::size_t attrLength, T const& sub) {
+  uint8_t* addInternal(char const* attrName, std::size_t attrLength, uint64_t tag, T const& sub) {
+    // When attrName is null add an array element instead -- simplifies code
+    // that builds both objects and arrays
+    if(attrName == nullptr) {
```

Comment:
The use case I have for this is when same code can generate both arrays and objects. I have a function for type conversion that has over 20 calls to add(Slice), add(Value) and add(ValuePair) and there it is very convenient to just pass a null key to signal that an array element should be added instead. Otherwise I would have to duplicate those calls.

But if you find that it is not reasonable I can live with one ugly function.

Reply:
I would prefer that function only to be called with with valid `attrName`s.